pkg/apis/hive/v1/clusterpool_types.go

```diff
@@ -28,6 +28,10 @@ type ClusterPoolSpec struct {
 	// ImageSetRef is a reference to a ClusterImageSet. The release image specified in the ClusterImageSet will be used
 	// by clusters created for this cluster pool.
 	ImageSetRef ClusterImageSetReference `json:"imageSetRef"`
+
+	// Labels to be applied to new ClusterDeployments created for the pool. ClusterDeployments already in the
+	//pool, or claimed from the pool, will not be affected when this value is modified.
```

Comment:
Is the "already in the pool" part long-term or just until we have the clusterpool controller replace ClusterDeployments on changes to the ClusterPool?

Reply:
If we pursue replacing every current pool on spec changes, this limitation will go away implicitly. I'll remove the note in documentation if we don't think it's permanent.